### PR TITLE
feat(telemetry): add telemetry config

### PR DESCRIPTION
**Pull request by lizradway**
## Description
- Merging [pre-approved PR](https://github.com/lizradway/sdk-typescript/pull/1) into feature branch rather than dev branch for tracking by team.
- Add `telemetry` singleton for OpenTelemetry configuration and setup.

This PR introduces the foundational telemetry configuration API for the Strands TypeScript SDK, following the Python SDK's design patterns. The `telemetry.setupTracer()` function provides a simple interface for configuring the tracer provider and exporters.

**Key features:**
- `setupTracer(config?)` - Configure tracer with optional provider and exporters
- OTLP exporter support (uses standard OTEL env vars: `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_HEADERS`)
- Console exporter for debugging
- Auto-flush on `beforeExit` for short-lived scripts
- NoOp pattern: When telemetry isn't configured, spans are automatically no-ops via OpenTelemetry's default `NoopTracerProvider` (parity with Python behavior)

**Usage:**
```typescript
import { telemetry } from '@strands-agents/sdk'

// Simple setup with OTLP exporter
telemetry.setupTracer({
  exporters: { otlp: true }
})

// With console exporter for debugging
telemetry.setupTracer({
  exporters: { otlp: true, console: true }
})

// Custom provider
telemetry.setupTracer({
  provider: new NodeTracerProvider({ resource: myResource }),
  exporters: { otlp: true }
})
```

This is PR 1 of a series that will add full telemetry support to the SDK.

## Related Issues

[#414](https://github.com/strands-agents/sdk-typescript/issues/414)

## Documentation PR

See [#421](https://github.com/strands-agents/sdk-typescript/issues/421) - Documentation will be added with the complete telemetry implementation

## Type of Change

New feature

## Testing

How have you tested the change?
- [x] I ran `npm run check`
- Integ test on PR failing due to AWS credentials on fork -- passed integ tests locally via `npm run test:integ`
- Unit tests added for all public API methods
- Tests cover OTLP configuration, console exporter, environment variable parsing, and edge cases

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [x] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,11 @@
     "@anthropic-ai/sdk": "^0.71.2",
     "@aws-sdk/client-secrets-manager": "^3.943.0",
     "@aws-sdk/credential-providers": "^3.943.0",
+    "@opentelemetry/api": "^1.9.0",
+    "@opentelemetry/exporter-trace-otlp-http": "^0.57.2",
+    "@opentelemetry/resources": "^1.30.1",
+    "@opentelemetry/sdk-trace-base": "^1.30.1",
+    "@opentelemetry/sdk-trace-node": "^1.30.1",
```
Comment (pgrayy) [inline package.json]: Nit: Alphabetize.
